h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -42,6 +45,9 @@
   String[][] _na_strings;       // Strings for NA in a given column
   String[][] _data;           // First few rows of parsed/tokenized data
 
+  int tentativeNumLines = -1;
+  int dataOffset = -1;
```

Comment:
My opinion is that tentativeNumLines & dataOffset should not be here. Two reasons: they only apply to the ARFF parser and they don't carry any meaningful information in the (unified) global setup (after two or more setups are merged in mergeSetups).